src/main/scala/is/hail/expr/Type.scala

```diff
@@ -115,6 +115,13 @@ sealed abstract class Type extends Serializable { self =>
       (TStruct.empty, a => null)
   }
 
+  def unsafeInsert(typeToInsert: Type, path: List[String]): (Type, UnsafeInserter) = {
+    if (path.nonEmpty)
+      TStruct.empty.unsafeInsert(typeToInsert, path)
+    else
+      (typeToInsert, (region, offset, rvb, inserter) => inserter())
```

Comment:
This duplicates the behavior in `TStruct.unsafeInsert`. It saves an allocation of `TStruct.empty`, but that doesn't happen per record, right? It's per-compilation.